src/command_modules/azure-cli-hdinsight/azure/cli/command_modules/hdinsight/custom.py

```diff
@@ -70,10 +67,10 @@ def create_cluster(cmd, client, cluster_name, resource_group_name, location=None
         cluster_configurations['gateway'] = dict()
     if 'restAuthCredential.isEnabled' not in cluster_configurations['gateway']:
         cluster_configurations['gateway']['restAuthCredential.isEnabled'] = 'true'
-    if not is_username_in_cluster_config:
-        cluster_configurations['gateway']['restAuthCredential.username'] = http_username
-    if not is_password_in_cluster_config:
-        cluster_configurations['gateway']['restAuthCredential.password'] = http_password
+    cluster_configurations['gateway']['restAuthCredential.username'] = http_username = \
+        http_username or cluster_configurations['gateway']['restAuthCredential.username']
+    cluster_configurations['gateway']['restAuthCredential.password'] = http_password = \
+        http_password or cluster_configurations['gateway']['restAuthCredential.password']
```

Comment:
I see what you're trying to do here. However, the chained assignment is not a very readable form of a statement, especially when the statement itself is long. It feels convoluted. To achieve the same effect, I suggest break each statement into two:
```Python
http_password = http_password or cluster_configurations['gateway']['restAuthCredential.password']
cluster_configurations['gateway']['restAuthCredential.password'] = http_password
```
The first allows the value of `http_password` to fall back.

It is definitely more verbose, however, we encourage readability.

Reply:
Sounds reasonable. Definitely let me know if there are more places that make you scratch your head (or if the intent isn't immediately obvious).